src/main/java/store/controller/StoreController.java

```diff
@@ -0,0 +1,137 @@
+package store.controller;
+
+import camp.nextstep.edu.missionutils.DateTimes;
+import java.util.ArrayList;
+import java.util.List;
+import store.domain.BuyProductParser;
+import store.domain.BuyProducts;
+import store.domain.FileReader;
+import store.domain.CustomerReceipt;
+import store.domain.Products;
+import store.domain.Promotions;
+import store.domain.Receipt;
+import store.domain.service.MemberShipCalculator;
+import store.domain.service.PromotionCalculator;
+import store.domain.service.PromotionChecker;
+import store.domain.vo.MemberShip;
+import store.view.InputView;
+import store.view.OutputView;
+import store.view.dto.request.PromotionRequest;
+import store.view.dto.response.ErrorResponse;
+import store.view.dto.request.MemberShipRequest;
+import store.view.dto.response.PromotionResponse;
+
+public class StoreController {
+
+    private static final String PRODUCTS_FILE_PATH = "/Users/moon/Desktop/java-convenience-store-7-moonwhistle/src/main/resources/products.md";
+    private static final String PROMOTION_FILE_PATH = "/Users/moon/Desktop/java-convenience-store-7-moonwhistle/src/main/resources/promotions.md";
+    private static final String YES = "Y";
+
+    private final MemberShipCalculator memberShipCalculator;
+    private final PromotionCalculator promotionCalculator;
+    private final OutputView outputView;
+    private final InputView inputView;
+
+    public StoreController(MemberShipCalculator memberShipCalculator, PromotionCalculator promotionCalculator,
+                           OutputView outputView, InputView inputView) {
+        this.memberShipCalculator = memberShipCalculator;
+        this.promotionCalculator = promotionCalculator;
+        this.outputView = outputView;
+        this.inputView = inputView;
+    }
+
+    public void run() {
+        boolean continueShopping = true;
+
+        Promotions promotions = getPromotions();
+        Products products = getProducts(promotions);
+
+        while (continueShopping) {
+            showInventory(products);
+            BuyProducts buyProducts = buyProducts(promotions, products);
+            CustomerReceipt customerReceipt = makeCustomerReceipt(products, buyProducts);
+            showReceipt(customerReceipt);
+            continueShopping = inputView.continueShopping()
+                    .equals(YES);
+        }
```

Comment:
다른 사람들의 코드를 몰래몰래 봤을때 이런씩으로 처리해서 변수를 줄이는 방법도 있더군요. 시각에 따라 좋을수도 있고 안좋다고 생각할 수도 있지만 코드보다가 생각나서 넣어둘게요

```suggestion
        do {
            showInventory(products);
            BuyProducts buyProducts = buyProducts(promotions, products);
            CustomerReceipt customerReceipt = makeCustomerReceipt(products, buyProducts);
            showReceipt(customerReceipt);
        } while (inputView.continueShopping()
                      .equals(YES))
```